### PR TITLE
[Codechange] Implements proper truck deletion in multiplayer

### PR DESCRIPTION
**Pull request by ulteq**
Fixes: #27

### CODE REVIEW

**Comment by only-a-ptr** [comment]
You're calling `NetworkStreamManager::getSingleton().removeStream()` from 2 places. It seems the call inside destructor is a safety net in case `calcPhysics()` fails to do it's job. That's not the way to go. Can't we come up with a reliable mechanism for deleting networked trucks?

Further, `NetworkStreamManager::getSingleton().removeStream()` looks like it might throw an exception. Beware: exception thrown in destructor = program terminated! http://stackoverflow.com/a/130123

**Comment by ulteq** [comment]
> You're calling `NetworkStreamManager::getSingleton().removeStream()` from 2 places.

Yes, for two different cases.

a) `state != NETWORKED` -> truck is local
* Truck is deleted here: https://github.com/RigsOfRods/rigs-of-rods/blob/688814c016b4f0abd85c398601aa6f147609709b/source/main/physics/BeamFactory.cpp#L701

b) `state == NETWORKED` -> truck is remote
* Truck is deleted here: https://github.com/RigsOfRods/rigs-of-rods/blob/688814c016b4f0abd85c398601aa6f147609709b/source/main/network/StreamableFactory.h#L169

> Beware: exception thrown in destructor = program terminated!

I've put it into the destructor, because the stream itself is added in the constructor:
https://github.com/RigsOfRods/rigs-of-rods/blob/master/source/main/physics/Beam.cpp#L1568

**Comment by only-a-ptr** [comment]
> Yes, for two different cases.

Oh, right, `(gEnv->networking != nullptr)` means "we're in multiplayer, each local truck has streams attached".

> I've put it into the destructor, because the stream itself is added in the constructor:

Hmm, as I can see, the whole network setup is plagued with the `if (!gEnv->network) return;` garbage. Anyway, bulky constructors with side effects, like `Beam()`, are wrong. The more we remove from there, the better. Factory methods (create/destroy) are the way to go.

File: source/main/physics/BeamData.h [hunk 1]
```diff
@@ -144,7 +144,6 @@ enum {
 	SLEEPING,       //!< not active, sleeping
 	NETWORKED,      //!< not calculated, gets remote data
 	NETWORKED_INVALID,	//!< not calculated, size differs from expected
-	RECYCLE,        //!< waiting for reusage
```
Comment (only-a-ptr) [inline source/main/physics/BeamData.h]: :100:

**Comment by only-a-ptr** [comment]
:+1:

**Pull request merged**